app/models/spree/stock/availability_validator_decorator.rb

```diff
@@ -0,0 +1,41 @@
+Spree::Stock::AvailabilityValidator.class_eval do
+  def validate(line_item)
+    # OFN specific check for in-memory :skip_stock_check attribute
+    return if line_item.skip_stock_check
+
+    quantity = adapt_line_item_quantity_to_inventory_units(line_item)
+    return if quantity == 0
+
+    validate_quantity(line_item, quantity)
+  end
+
+  private
+
+  # This is an adapted version of a fix to the inventory_units not being considered here.
+  # See #3090 for details.
+  # This can be removed after upgrading to Spree 2.4.
+  def adapt_line_item_quantity_to_inventory_units(line_item)
```

Comment:
What do you think about this name?
```suggestion
  def unshipped_quantity(line_item)
```

Reply:
thanks for this. I ended up changing the method to return a quantity and then called it quantity_in_shipment, because inventory_units can be already shipped as well.